### PR TITLE
[Auth] Convert LoginView to SwiftUI

### PR DESCRIPTION
**Pull request by ncooke3**
New UI for email/password page:

Screenshot 2024-12-12 at 11 28 29 AM

#no-changelog

### CODE REVIEW

File: FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift [hunk 1]
```diff
@@ -12,162 +12,179 @@
...
+        TextField("Password", text: $password)
+          .textFieldStyle(SymbolTextField(symbolName: "lock.fill"))
+        Spacer()
+        Group {
+          LoginViewButton(
+            text: "Login",
+            accentColor: .white,
+            backgroundColor: .orange
+          ) {
+            Task {
```
Comment (peterfriese) [inline FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift:64]: I'd refactor the code in this closure into a function on the view. Or maybe even into a view model.

File: FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift [hunk 2]
```diff
@@ -12,162 +12,179 @@
...
+                } else {
+                  // TODO(ncooke3): Implement handling of other MFA provider (phone).
+                }
+              } catch {
+                print(error.localizedDescription)
+              }
+            }
+          }
+          .alert("Enter one time passcode.", isPresented: $showingAlert) {
+            TextField("Verification Code", text: $onetimePasscode)
```
Comment (peterfriese) [inline FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift:94]: To keep the view's `body` readable, this should probably be a standalone view or a view builder function on the surrounding view

File: FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift [hunk 3]
```diff
@@ -12,162 +12,179 @@
...
-  }
-
-  private func setupCreateAccountButton() {
-    addSubview(createAccountButton)
-    createAccountButton.translatesAutoresizingMaskIntoConstraints = false
-    NSLayoutConstraint.activate([
-      createAccountButton.centerXAnchor.constraint(equalTo: centerXAnchor),
-      createAccountButton.topAnchor.constraint(equalTo: loginButton.bottomAnchor, constant: 5),
-    ])
+// TODO(ncooke3): Use view modifiers?
```
Comment (peterfriese) [inline FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift:165]: Or implement a custom button style.

Comment (ncooke3) [reply FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift:165]: Yes, a button style sounds good. When I tried to create a very simple button style, the custom style lost the button's default ability to change text color on tap. I feel like I'm not doing something right.

```swift
private struct MyButtonStyle : ButtonStyle {
  func makeBody(configuration: Configuration) -> some View {
    configuration.label
  }
}

#Preview {
  Button("Hello") {}
  .buttonStyle(MyButtonStyle())
}
```

Screenshot 2024-10-14 at 12 59 42 PM

Should I leave this as is? What's the most idiomatic pattern?

File: FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift
Comment (ncooke3) [inline FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift:1]: Current render:
Screenshot 2024-10-14 at 1 03 50 PM

Any feedback welcome.

File: FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift (hunk 1 above)
Comment (ncooke3) [reply FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift]: Refactored out into a function on the view. I'll leave this thread open to consider doing a view model. For now, I'll focus on refactoring the UI bits..

File: FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift (hunk 2 above)
Comment (ncooke3) [reply FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift]: Hmm, sorry could you explain this a little bit more? I'm not fully following.

File: FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift [hunk 4]
```diff
@@ -12,162 +12,145 @@
...
-      ),
-      passwordTextField.heightAnchor.constraint(equalToConstant: 45),
-    ])
-
-    let constant: CGFloat = UIDevice.current.orientation.isLandscape ? 5 : 20
-    passwordTopConstraint =
-      passwordTextField.topAnchor.constraint(
-        equalTo: emailTextField.bottomAnchor,
-        constant: constant
+extension LoginView {
```
Comment (ncooke3) [inline FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift:81]: Screenshot 2024-12-12 at 11 28 29 AM

Comment (ncooke3) [reply FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift:92]: Putting the UI code in an extension to keep all the UI code in one place.

**Review by paulb777 — APPROVED** [review]

File: FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift (hunk 4 above)
Comment (ncooke3) [reply FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift:92]: Resolving so I can merge.

**Comment by ncooke3** [comment]
Will address any further feedback in a following PR.

**Pull request merged**